woocommerce/woocommerce-ios
# [Subscription products] Show subscription product types in "Add product" bottom sheet

Closes: #11116 


## Description

Starts showing the Subscription product types in the Product type bottom sheet. 

Changes
- New eligibility checker to check for Woo Subscriptions plugin eligibility. 
- Update `BottomSheetProductType` to support new cases for subscription products.
- Show subscription product types in the Product type bottom sheet after checking eligibility. 

## Testing instructions

1. Enable `subscriptionProducts` feature flag locally by returning `true` from this line https://github.com/woocommerce/woocommerce-ios/pull/11118/files#diff-26c9afc49bb1f1a168517d43d5e36a64ff45adb046469369ef3e4d6800ccc19bR88
1. Install the [WooCommerce Subscriptions](https://woo.com/products/woocommerce-subscriptions) plugin on a site
1. Login into the app
1. Navigate to Products tab
1. Tap on the `+` button from top right corner to start the add new product flow
1. Tap on the "Add manually" option if needed
1. Notice the bottom sheet now contains two new subscription product types
1. Switch to a site without the WooCommerce Subscriptions plugin.
1. Repeat steps 3-5
1. Notice the bottom sheet doesn't contain subscription product types.

## Screenshots

| Woo Subscriptions plugin installed and active | Woo Subscriptions plugin not available |
|--------|--------|
| ![Simulator Screenshot - iPhone 14 Pro - 2023-11-09 at 09 05 53](https://github.com/woocommerce/woocommerce-ios/assets/524475/8de7bcb8-a416-479b-b1aa-be91008a2a67) | ![Simulator Screenshot - iPhone 14 Pro - 2023-11-09 at 09 06 17](https://github.com/woocommerce/woocommerce-ios/assets/524475/286a8720-14d6-40c5-8d30-c3c854dfbdfb) | 

---
